crates/guest/assessor/assessor-guest/src/main.rs

```diff
@@ -63,24 +60,34 @@ fn main() {
         } else {
             fill.verify_signature(&eip_domain_separator).expect("signature does not verify")
         };
-        fill.evaluate_requirements().expect("requirements not met");
-        env::verify_integrity(&fill.receipt_claim()).expect("claim integrity check failed");
-        let claim_digest = fill.receipt_claim().digest();
+        let claim_digest = fill.evaluate_requirements().expect("requirements not met");
         let commit = AssessorCommitment {
             index: U256::from(index),
             id: fill.request.id,
             requestDigest: request_digest.into(),
             claimDigest: <[u8; 32]>::from(claim_digest).into(),
+            fulfillmentDataDigest: <[u8; 32]>::from(
+                fill.fulfillment_data
+                    .fulfillment_data_digest()
+                    .expect("failed to get fulfillment data digest"),
+            )
+            .into(),
         }
         .eip712_hash_struct();
         leaves.push(Digest::from_bytes(*commit));
-        if fill.request.requirements.callback.addr != Address::ZERO {
-            callbacks.push(AssessorCallback {
-                index: index.try_into().expect("callback index overflow"),
-                addr: fill.request.requirements.callback.addr,
-                gasLimit: fill.request.requirements.callback.gasLimit,
-            });
+
+        let callback = &fill.request.requirements.callback;
+
+        if let FulfillmentData::ImageIdAndJournal(_, _) = &fill.fulfillment_data {
+            if callback.addr != Address::ZERO {
+                callbacks.push(AssessorCallback {
+                    index: index.try_into().expect("callback index overflow"),
+                    addr: callback.addr,
+                    gasLimit: callback.gasLimit,
+                });
+            }
         }
```

Comment:
As discussed on the call, it would be good to have the assessor fail if a callback is requested but cannot be completed. As is, the callback would actually be skipped which means the client would request a callback and not get it.
```suggestion
        if callback.addr != Address::ZERO {
            match &fill.fulfillment_data {
                 FulfillmentData::ImageIdAndJournal(_, _) => {}
                 _ => panic!("callback requested but no fulfillment data provided");
            }
            callbacks.push(AssessorCallback {
                index: index.try_into().expect("callback index overflow"),
                addr: callback.addr,
                gasLimit: callback.gasLimit,
            });
        }
```